transport/TcpServerEndpoint.h

```diff
@@ -46,7 +48,7 @@ class TcpServerEndpoint : public ServerEndpoint, public RtcePoll::IEventListener
 
 public: // internal job methods
```

Comment:
Why internal job's method are public?

Reply:
I see. I would definitely prefer combining all job-called "internal" methods into an interface, and pass it to the job.

Pros: clearly readable code and intent. Speed does not change. Also, combining them into interfaces would make more prominent similarities (and allow to combine and thus reduce codebase size).
Cons: codebase size slightly increases.